### PR TITLE
Add source location and checksum in generated JSON data.

### LINKED ISSUE #804: Generated JSON data does not contain any link to the sources of the packages.
This is quite unfortunate as it does not allow to solve the issue that was raised in this thread: https://coq.discourse.group/t/coqhammer-1-1-1-for-coq-8-9/327.

The CoqGym (@yangky11) people crawled the Coq package index to get the URLs of packages.
Now they could obtain exactly the same results by doing `cat coq-packages.json | jq '.[][1].versions[0].homepage'`, which is already an improvement.
However, it doesn't solve their bug which is to confuse the homepage of a package and the repository URL. It just happened that this confusion gave them correct results in most cases, but not for instance for math-classes and corn (as remarked by @spitters).

Would it be doable to add more meta-information to reflect, both the repository as declared in the opam meta-data, but also the URL of sources, which is defined there as well?

### PR DESCRIPTION
**Pull request by maximedenes**
Fixes #804

### CODE REVIEW

**Comment by maximedenes** [comment]
@gares does this look good to you?

File: scripts/archive2web.ml [hunk 1]
```diff
@@ -103,6 +109,11 @@ let parse_author_name s =
     Str.matched_string s
   else raise (Invalid_argument "author_name")
 
+let obind o f =
```
Comment (gares) [inline scripts/archive2web.ml]: `Option.map` from https://caml.inria.fr/pub/docs/manual-ocaml/libref/Stdlib.Option.html

Comment (gares) [reply scripts/archive2web.ml]: This is since 4.08, hence my comment

Comment (maximedenes) [reply scripts/archive2web.ml]: Yeah, I used that initially. Then I hit some issues with our CI not using 4.08, I fixed it. Then I hit issues with Nix not packaging an 4.08-compatible version of `ppx_deriving_yojson`. I fixed it. Then I hit an issue with the nix channel not being up-to-date yet. Then I thought I'd rather define the function.

And btw, this is `bind`, not `map`, hence the name ;)

Comment (gares) [reply scripts/archive2web.ml]: Hem, so, are you on 4.8? Cause the pr says so. Anyway the changes are trivial, they can be merged I guess

Comment (maximedenes) [reply scripts/archive2web.ml]: > Hem, so, are you on 4.8? Cause the pr says so. Anyway the changes are trivial, they can be merged I guess

Ah sorry! I forgot to update the PR message when I switched back to 4.06, now I understand the confusion.

**Pull request merged**